lib/rules/naming/style-guide-casing.js

```diff
@@ -178,7 +178,14 @@ class StyleGuideCasingChecker extends BaseChecker {
   }
 
   FunctionDefinition(node) {
-    if (!isMixedCase(node.name) && !node.isConstructor) {
+    if (node.isConstructor) return
+
+    // Allow pure functions to be either mixedCase or UPPER_SNAKE_CASE
+    if (node.stateMutability === 'pure') {
```

Comment:
this `if` should also include a check for the function being inside an interface, as to not unnecessarily stray too far from the official style guide. Keep in mind that [free functions](https://docs.soliditylang.org/en/latest/contracts.html#index-8) are a thing (not widely used thing, but still :D) and add tests accordingly. You can probably make this easier by using the `findFirstParent` helper function